frontend/src/metabase/query_builder/components/QueryDownloadWidget.jsx

```diff
@@ -69,11 +72,23 @@ const SavedQueryButton = ({ className, type, result: { json_query }, card }) =>
         {type}
     </DownloadButton>
 
+/// Parse the query string part of the URL (e.g. the `?key=value` part) into an object. We need to pass them this way
+/// to the `DownloadButton` because it's a form which means we need to insert a hidden `<input>` for each param we
+/// want to pass along.
+/// In this case the params are QP params -- template tags and the like.
+///
+/// TODO - there's probably some better way of doing this or some better place to put this code. Open to suggestions.
+function windowQueryParams() {
+    const query = urlParse(window.location.href).query; // get the part of the URL that looks like key=value
+    return query && querystring.parse(query);           // expand them out into a map
+}
```

Comment:
open to better ways of doing this or a better place to put this function.

I know with React/Redux you're not supposed to do anything the easy way any more so I'm sure instead of doing `window.location.href` I'm probably supposed to define a new class and inject the React router with `@withRouter` and some other ridiculous nonsense. I couldn't figure out how tho do that but if someone wants to point the way I'm all ears